alicat/serial.py

```diff
@@ -116,37 +116,173 @@ def get(self, retries=2):
                 'N2O', 'Ne', 'O2', 'C3H8', 'n-C4H10', 'C2H2', 'C2H4',
                 'i-C2H10', 'Kr', 'Xe', 'SF6', 'C-25', 'C-10', 'C-8', 'C-2',
                 'C-75', 'A-75', 'A-25', 'A1025', 'Star29', 'P-5'
+
+                Gas mixes may only be called by their mix number.
         """
         self._test_controller_open()
 
-        if gas not in self.gases:
+        if isinstance(gas, int):
+            return self._set_gas_number(gas, retries)
+        else:
+            return self._set_gas_name(gas, retries)
+
+    def _set_gas_number(self, number, retries):
+        self._test_controller_open()
+        command = '{addr}$${index}\r'.format(addr=self.address,
+                                             index=number)
+        self._write_and_read(command, retries)
+
+        reg46 = self._write_and_read('{addr}$$R46\r'.format(
+            addr=self.address
+        ), retries)
+        reg46_gasbit = int(reg46.split()[-1]) & 0b0000000111111111
+
+        if number != reg46_gasbit:
+            raise IOError("Cannot set gas.")
+        pass
+
+    def _set_gas_name(self, name, retries):
+        self._test_controller_open()
+        if name not in self.gases:
             raise ValueError("{} not supported!".format(gas))
-        command = '{addr}$${gas}\r'.format(addr=self.address,
-                                           gas=self.gases.index(gas))
+        command = '{addr}$${gas}\r'.format(
+            addr=self.address,
+            gas=self.gases.index(name)
+        )
+        self._write_and_read(command, retries)
+
+        reg46 = self._write_and_read('{addr}$$R46\r'.format(
+            addr=self.address
+        ), retries)
+        reg46_gasbit = int(reg46.split()[-1]) & 0b0000000111111111
+
+        if self.gases.index(name) != reg46_gasbit:
+            raise IOError("Cannot set gas.")
+        pass
+
+    def create_mix(self, mix_no, name, gases, retries=2):
+        """Create a gas mix.
+
+        Gas mixes are made using COMPOSER software.
+        COMPOSER mixes are only allowed for firmware 5v or greater.
+
+        Args:
+        mix_no: The mix number. Gas mixes are stored in slots 236-255.
+        name: A name for the gas that will appear on the front panel.
+        Names greater than six letters will be cut off.
+        gases: A dictionary of the gas by name along with the associated
+        percentage in the mix.
+        """
+        self._test_controller_open()
+
+        read = '{addr}VE\r'.format(addr=self.address)
+        firmware = self._write_and_read(read, retries)
+        if "2v" in firmware\
+                or "3v" in firmware\
+                or "4v" in firmware\
+                or "GP" in firmware:
+            raise IOError("This unit does not support COMPOSER gas mixes.")
+
+        if mix_no < 236 or mix_no > 255:
+            raise ValueError("Mix number must be between 236-255!")
+
+        gas = [str(i) for i in gases.keys()]
+        percent = [str(i) for i in gases.values()]
```

Comment:
Do you use these anywhere? I think they can be deleted because you use `gases` below?

Reply:
I suppose not. I will remove them.